### ISSUE TITLE
Pi-Hole - Error when updating to version 2.0

### ISSUE THREAD

**Issue by chaoscode** [truenas/charts#2169]
When I updated today to version 2.0, I got a loop error in the logs saying:

"pihole-FTL: no process found"

When I rolled back to 1.0.25, the error went away.

TrueNAS Scale Version:
TrueNAS-SCALE-23.10.1

**Comment by chaoscode** [comment]
tried to update TrueNAS Scale to Version: TrueNAS-SCALE-23.10.1.3 but same issue. Just loops "pihole-FTL: no process found"

No other errors in the log.

**Comment by TuKo1982** [comment]
Same error here on TrueNAS-SCALE-23.10.1.3

**Comment by PrinzJuliano** [comment]
It seems to be a permissions error binding to port 53:
Pui-hole diagnosis 
DNSMASQ_CONFIG
FTL FAILED TO START DUE TO FAILED TO CREATE LISTENING SOCKET FOR PORT 53: PERMISSION DENIED

**Comment by johnestep** [comment]
Same issue here

**Comment by johnestep** [comment]
> When I updated today to version 2.0, I got a loop error in the logs saying:
> 
> "pihole-FTL: no process found"
> 
> When I rolled back to 1.0.25, the error went away.
> 
> TrueNAS Scale Version: TrueNAS-SCALE-23.10.1

SAME ISSUE HERE AS WELL.

**Comment by ccMatrix** [comment]
FTL is still not starting with the 2.0.1 release.

**Comment by stavros-k** [comment]
> FTL is still not starting with the 2.0.1 release.

Well, that's embarrassing :D
It indeed does not start, but it did while testing, let me take another look.

**Comment by ccMatrix** [comment]
No worries, I'm glad there is a rollback feature which works perfectly :)

**Comment by stavros-k** [comment]
Well, I didn't sync my testing fixes in the actual fix. :D

Should be okay now after #2175 !

Thanks

**Comment by chaoscode** [comment]
> > FTL is still not starting with the 2.0.1 release.
> 
> Well, that's embarrassing :D It indeed does not start, but it did while testing, let me take another look.

No worries, the best thing about docker is the rollback feature. 👍 

As soon as 2.0.2 is released, I will check it out and confirm.

**Comment by ccMatrix** [comment]
Updated to 2.0.2 but still no FTL starting up for me.

The logs showed these lines that triggered an investigation:
```
2024-02-16 14:39:46.397425+01:00[i] Number of gravity domains: 1155976 ( unique domains)
2024-02-16 14:39:46.398820+01:00Error: in prepare, near ")": syntax error
2024-02-16 14:39:46.398847+01:00NTO info (property,value) VALUES ('gravity_count',);
2024-02-16 14:39:46.398855+01:00error here ---^
```

I logged into the container and found that running `/opt/pihole/gravity.sh` failed with an error:
```
  [✗] Unable to build gravity tree in /etc/pihole/gravity.db_temp
  
   [✗] Unable to create database. Please contact support.
```

I've checked the files in `/etc/pihole` and found the following files:
```
root@truenas:/etc/pihole# ls -la|grep gravity
-rw-rw-r-- 1 pihole pihole  77889536 Feb 16 15:07 gravity.db
-rw-r--r-- 1 root   root    43544576 Feb 16 15:08 gravity.db_temp
-rw-r--r-- 1 root   root        9728 Feb 16 15:08 gravity.db_temp-journal
```

So while gravity.db was owned by user pihole with permissions 664 the files owned by root where 644. I changed the permissions to 664 via `chmod 664 /etc/pihole/gravity.db_temp*` and then ran the gravity.sh again this time it seemingly succeeded.
```
-rw-rw-r-- 1 pihole pihole  59641856 Feb 16 15:08 gravity.db
-rw-rw-r-- 1 pihole pihole  77889536 Feb 16 15:07 gravity_old.db
```

It looks like the migration of the old gravity.db to the new one might be running as the wrong user or the file is created with the wrong permissions for it to succeed thereby blocking the startup process.

But now logging in into the UI I am unfortunately greeeted by the following Pi-hole diagnosis error:
![image](https://github.com/truenas/charts/assets/96917/15908e46-291b-4c1e-823a-26a59134f49c)

So this capability seems to still be necessary and missing in 2.x in addition to the migration issue.

**Comment by ccMatrix** [comment]
So the migration thing might be somehow unique to my install. But I can work around that with the steps above. Looking forward to 2.0.3 for testing

**Comment by AFellmett** [comment]
Updated it from 1.0.25 to 2.0.2 and it's now working as before without any issues.
Also run the manual `update gravity` command, without any issues.

Thx @stavros-k

**Comment by ccMatrix** [comment]
I've setup a secondary pihole instance of 2.0.2 to test and it looks like the NET_RAW is needed when pihole is running as a DHCP server. 

@AFellmett Are you using pihole for DNS only or also as a DHCP server?

**Comment by AFellmett** [comment]
@ccMatrix that's a good point, I use it only for DNS not DHCP.

When I activate the DHCP Option through pihole webui, I get the same issue regarding `NET_RAW`.
But I didn't check if that's still the case if I `Enable DHCP` within the Truenas chart configuration.

![image](https://github.com/truenas/charts/assets/17900799/3efc7c96-4f0d-4d75-9dee-6581f5349e3e)

**Comment by ccMatrix** [comment]
From what I see in the configuration files here: https://github.com/truenas/charts/blob/308b345ab6e94ccb673f34094cef2ada8cbec1bc/charts/pihole/2.0.2/templates/_pihole.tpl#L36

It will enable DHCP in Pi-Hole with the values entered so it starts up with DHCP enabled.

**Comment by chaoscode** [comment]
I do want to give a shoutout to everyone for helping to solve this. Nice to see people working together and not posting in all caps.  Massive props to the testers and devs.

**Comment by AFellmett** [comment]
I'm sorry to point out another issue I came across.
Unfortunately the migration process cleared my list of CNAME records within the local dns section of pi-hole. 
I had no issues with the dns records itself.
![image](https://github.com/truenas/charts/assets/17900799/982b0647-d062-4912-b414-725c85c9f625)

I know, that these information were stored in different files in earlier versions.

**Comment by stavros-k** [comment]
> I'm sorry to point out another issue I came across.
> 
> Unfortunately the migration process cleared my list of CNAME records within the local dns section of pi-hole. 
> 
> I had no issues with the dns records itself.
> 
> ![image](https://github.com/truenas/charts/assets/17900799/982b0647-d062-4912-b414-725c85c9f625)
> 
> 
> 
> I know, that these information were stored in different files in earlier versions.

Found the issue, sorry for that.
(#2185)
It should fix it self on the next update

**Comment by stavros-k** [comment]
> I'm now getting an error when trying to upgrade. I think it might be related to all the rollbacks I had to do? Haven't really had a chance to go through the migration script myself.

Whops. changed the order a bit here to fix that. https://github.com/truenas/charts/pull/2180/commits/b50a594010e539a314f97d91e9c05cc4957cc81d

**Comment by stavros-k** [comment]
> Update from 1.0.25 to 2.0.4 now worked for me. FTL is running, no errors in migration.
> 
> Thanks for the update. Is this a preparation to later update to the latest pihole docker image more easily? (2023.11.0 to e.g. 2024.02.0) ?

https://github.com/truenas/charts/pull/2195
This should fix it
